### PR TITLE
[WIP] Fir example

### PR DESCRIPTION
**Pull request by bthirion**
This PR adds an example with a Finite Impulse Response (FIR) model, which is a non-parametric way of fitting the hemodynamic response in fMRI.
Took this opportunity to clean some technical details, like the oversampling factor that should not be 1.

### CODE REVIEW

File: examples/02_first_level_models/plot_fir_model.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#########################################################################
+# Next solution is to try fininte impulse reponse (FIR) models: we just say that the hrf is an arbitrary function that lags behind the stimulus onset.
+# In the present case, given that the numbers of condition is high, we should use a simple FIR model.
+# 
+# Concretely, we set `hrf_model` to 'fir' and `fir_delays` to [3, 5, 7] (s)
+
+from nistats.first_level_model import FirstLevelModel
+from nistats.reporting import plot_design_matrix, plot_contrast_matrix
+
+first_level_model = FirstLevelModel(t_r, hrf_model='fir', fir_delays=[1, 2, 3])
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: does `fir_delays=[1, 2, 3]` correspond to 3, 5, 7s as stated above? What is the unit?

File: examples/02_first_level_models/plot_fir_model.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+Specifically, the so-called 'localizer' dataset proposed by Pinel is
+downloaded, then analysed with a FIR model with 3 lags.
+4 main contrasts are estimated.
+
+"""
+
+#########################################################################
+# Grab localizer data
+
+import pandas as pd
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: If you delete the unused sections containing the functions  `make_localizer_contrasts` and  `plot_contrast`, you could put move all imports here:
```python
import numpy as np
import pandas as pd
from nistats import datasets
from nilearn import plotting
import matplotlib.pyplot as plt
```

File: examples/02_first_level_models/plot_fir_model.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
+"""Analysis of an fMRI dataset with a Finite Impule Response (FIR) model
+=====================================================================
+
+FIR models are used to estimate the hemodyamic response non-parametrically.
+The example below shows that they're good to do statistical inference
+even on fast event-related fMRI datasets.
+
+Specifically, the so-called 'localizer' dataset proposed by Pinel is
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: Maybe replace by "Here, we demonstrate the use of a FIR model with 3 lags, computing 4 contrasts from a single subject dataset from the "Neurospin Localizer". It is a fast event related design: During 5 minutes, 80 events of the following types are presented : ['calculaudio', 'calculvideo'', 'clicDvideo', 'clicGaudio', 'clicGvideo', 'damier_H', 'damier_V', 'phraseaudio', 'phrasevideo']"

File: examples/02_first_level_models/plot_fir_model.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+t_r = 2.4
+events_file = data['events']
+events = pd.read_table(events_file)
+
+#########################################################################
+# Automate contrast computation
+
+import numpy as np
+
+def make_localizer_contrasts(design_matrix):
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: delete?

File: examples/02_first_level_models/plot_fir_model.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    }
+    return contrasts
+
+#########################################################################
+# Result plotting function
+
+from nilearn import plotting
+import matplotlib.pyplot as plt
+
+def plot_contrast(first_level_model):
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: delete?

File: examples/02_first_level_models/plot_fir_model.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    "H-V": contrasts["damier_H"] - contrasts["damier_V"],
+    "audio-video": contrasts["audio"] - contrasts["video"],
+    "computation-sentences": (contrasts["computation"] -
+                              contrasts["sentences"]),
+    }
+
+plot_contrast_matrix(contrasts['left-right'], design_matrix)
+
+#########################################################################
+# Take a breathe.
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: I had to take a breath, indeed.

File: examples/02_first_level_models/plot_fir_model.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+for index, (contrast_id, contrast_val) in enumerate(contrasts.items()):
+    ax = plt.subplot(1, len(contrasts), 1 + index)
+    z_map = first_level_model.compute_contrast(
+        contrast_val, output_type='z_score')
+    plotting.plot_stat_map(
+        z_map, display_mode='z', threshold=3.0, title=contrast_id, axes=ax,
+        cut_coords=1)
+
+#########################################################################
+# The result is not convincing to my eyes. Maybe we're asking a bit too much to a small dataset, with a relatively large number of experimental conditions!
```
Comment (chrplr) [inline examples/02_first_level_models/plot_fir_model.py]: No that bad. Only H-V is odd to me.

**Comment by bthirion** [comment]
@chrplr I think I ahve addressed your comments

**Comment by bthirion** [comment]
Any more review on this ?

**Comment by bthirion** [comment]
Any other review ?

**Comment by bthirion** [comment]
Will merge tomorrow unless I hear otherwise.
B

**Pull request merged**